src/keyring/KeyringController.ts

```diff
@@ -416,6 +416,14 @@ export class KeyringController extends BaseController<KeyringConfig, KeyringStat
     return privates.get(this).keyring.store.unsubscribe(listener);
   }
 
+  onLock(listener: Listener<KeyringState>) {
+    return privates.get(this).keyring.on('lock', listener);
+  }
+
+  onUnlock(listener: Listener<KeyringState>) {
+    return privates.get(this).keyring.on('unlock', listener);
+  }
+
```

Comment:
I think these listeners should be `() => void`, not `Listener<KeyringState>`.

Reply:
Oh yes I copy pasted to test it quickly and forgot to change. Also still need to insert some comments on the functions and create the tests.